tests/components/upnp/test_config_flow.py

```diff
@@ -119,6 +127,60 @@ async def test_flow_config(hass: HomeAssistantType):
         assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
         assert result["title"] == mock_device.name
         assert result["data"] == {
-            "st": mock_device.device_type,
-            "udn": mock_device.udn,
+            CONFIG_ENTRY_ST: mock_device.device_type,
+            CONFIG_ENTRY_UDN: mock_device.udn,
+        }
+
+
+async def test_options_flow(hass: HomeAssistantType):
+    """Test options flow."""
+    # Set up config entry.
+    udn = "uuid:device_1"
+    mock_device = MockDevice(udn)
+    discovery_infos = [
+        {
+            DISCOVERY_UDN: mock_device.udn,
+            DISCOVERY_ST: mock_device.device_type,
+            DISCOVERY_LOCATION: "http://192.168.1.1/desc.xml",
         }
+    ]
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONFIG_ENTRY_UDN: mock_device.udn,
+            CONFIG_ENTRY_ST: mock_device.device_type,
+        },
+        options={CONFIG_ENTRY_SCAN_INTERVAL: DEFAULT_SCAN_INTERVAL},
+    )
+    config_entry.add_to_hass(hass)
+
+    config = {
+        # no upnp, ensures no import-flow is started.
+    }
+    async_discover = AsyncMock(return_value=discovery_infos)
+    with patch.object(
+        Device, "async_create_device", AsyncMock(return_value=mock_device)
+    ), patch.object(Device, "async_discover", async_discover):
+        # Initialisation of component.
+        await async_setup_component(hass, "upnp", config)
+        await hass.async_block_till_done()
+
+        # DataUpdateCoordinator gets a default of 30 seconds for updates.
+        coordinator = hass.data[DOMAIN]["coordinators"][mock_device.udn]
+        assert coordinator.update_interval == timedelta(seconds=DEFAULT_SCAN_INTERVAL)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)